emukit/bayesian_optimization/loops/bayesian_optimization_loop.py

```diff
@@ -15,22 +19,19 @@
 
 
 class BayesianOptimizationLoop(OuterLoop):
-    def __init__(self, model: IModel, space: ParameterSpace, X_init: np.array, Y_init: np.array,
-                 acquisition: Acquisition = None, candidate_point_calculator: CandidatePointCalculator = None,
+    def __init__(self, model: IModel, space: ParameterSpace, acquisition: Acquisition = None, batch_size: int = 1,
                  model_updater: ModelUpdater = None):
 
         """
         Emukit class that implement a loop for building modular Bayesian optimization
 
         :param model: The model that approximates the underlying function
         :param space: Input space where the optimization is carried out.
-        :param acquisition: The acquisition function that will be used to collect new points (default, EI).
-        :param X_init: x values for initial function evaluations
-        :param Y_init: y values for initial function evaluations
-        :param model_updater: Defines how and how often the model will be updated if new data
-        arrives (default, FixedIntervalUpdater)
-        :param candidate_point_calculator: Optimizes the acquisition function to find the
-        next candidate to evaluate (default, Sequential)
+        :param acquisition: The acquisition function that will be used to collect new points (default, EI). If batch
+                            size is greater than one, this acquisition must output positive values only. 
+        :param batch_size: How many points to evaluate in one iteration of the optimization loop. Defaults to 1.
+        :param model_updater: Defines how and when the model is updated if new data arrives.
```

Comment:
Not in this PR, but I now fully agree with the point of not surfacing our loop step abstractions on this level. Thus we should consider getting rid of this ModelUpdater argument here. Created #63 for that